File: MATTHEW/leetcode/AIcoding/3.DeepSeek-v3/go_solutions/914-x-of-a-kind-in-a-deck-of-cards.go
func hasGroupsSizeX(deck []int) bool {
    freq := make(map[int]int)
    for _, num := range deck {
        freq[num]++
    }

    g := -1
    for _, count := range freq {
        if g == -1 {
            g = count
        } else {
            g = gcd(g, count)
        }
    }

    return g >= 2
}

func gcd(a, b int) int {
    for b != 0 {
        a, b = b, a % b
    }
    return a
}

File: MATTHEW/leetcode/AIcoding/5.Qwen-3/python3_solutions/914-x-of-a-kind-in-a-deck-of-cards.py
from collections import Counter
import math

class Solution:
    def hasGroupsSizeX(self, deck: List[int]) -> bool:
        count = Counter(deck)
        min_freq = min(count.values())
        for i in range(2, min_freq + 1):
            if all(freq % i == 0 for freq in count.values()):
                return True
        return False